components-core/src/test/java/com/adyen/checkout/components/core/internal/util/DateUtilsTest.kt

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ararat on 22/1/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.util
+
+import com.adyen.checkout.core.AdyenLogger
+import com.adyen.checkout.core.internal.util.Logger
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.Calendar
+import java.util.Locale
+
+class DateUtilsTest {
+
+    @BeforeEach
+    fun beforeEach() {
+        AdyenLogger.setLogLevel(Logger.NONE)
+    }
+
+    @ParameterizedTest
+    @MethodSource("parseDateToView")
+    fun testParseDateToView(month: String, year: String, expectedFormattedDate: String) {
+        val formattedDate = DateUtils.parseDateToView(month, year)
+
+        assertEquals(expectedFormattedDate, formattedDate)
+    }
+
+    @ParameterizedTest
+    @MethodSource("toServerDateFormat")
+    fun testToServerDateFormat(calendar: Calendar, expectedFormattedDate: String) {
+        val formattedDate = DateUtils.toServerDateFormat(calendar)
+
+        assertEquals(expectedFormattedDate, formattedDate)
+    }
+
+    @ParameterizedTest
+    @MethodSource("matchesFormat")
+    fun testMatchesFormat(date: String, format: String, expectedResult: Boolean) {
+        val result = DateUtils.matchesFormat(date, format)
+
+        assertEquals(expectedResult, result)
+    }
+
+    @ParameterizedTest
+    @MethodSource("formatStringDate")
+    fun testFormatStringDate(date: String, shopperLocale: Locale, inputFormat: String, expectedFormattedDate: String?) {
+        val formattedDate = DateUtils.formatStringDate(date, shopperLocale, inputFormat)
+
+        assertEquals(expectedFormattedDate, formattedDate)
+    }
+
+    @ParameterizedTest
+    @MethodSource("formatDateToString")
+    fun testFormatDateToString(calendar: Calendar, pattern: String, expectedFormattedDate: String?) {
+        val formattedDate = DateUtils.formatDateToString(calendar, pattern)
+
+        assertEquals(expectedFormattedDate, formattedDate)
+    }
+
+    companion object {
+        @JvmStatic
+        fun parseDateToView() = listOf(
+            // month, year, expectedFormattedDate
+            Arguments.arguments("1", "2021", "1/21"),
```

Comment:
Nit: if you import `arguments()` directly it's possible to omit the verbose `Arguments.` part